src/schema/posts.ts

```diff
@@ -1643,6 +1662,35 @@ export const resolvers: IResolvers<any, Context> = {
       );
       return getPostById(ctx, info, newPost.id);
     },
+    editSharePost: async (
+      _,
+      {
+        id,
+        sourceId,
+        commentary,
+      }: { id: string; sourceId: string; commentary: string },
+      ctx,
+      info,
+    ): Promise<GQLPost> => {
+      const post = await ctx.con.getRepository(Post).findOneByOrFail({ id });
+
+      if (post.authorId !== ctx.userId) {
+        throw new ForbiddenError(
+          `Editing other people's posts is not allowed!`,
```

Comment:
Since this is now re-used logic, can we maybe either abstract it to a function or use a enum for the error message?
(in case of changing the message for instance, but option 1 would be my preference)

Reply:
I started extracting this into a function, but realized in the process that this is not entirely re-used logic, the way it is used in `editPost` is based on slightly different conditions. 

The exception management could use some structure here, but it feels like a more involved refactor that could be done separately. What are your thoughts?

Reply:
Ah ok then never mind, it's more a nice to have for now.